### PR TITLE
neonvm-controller: refactor doReconcile

### PR DESCRIPTION
**Pull request by Omrigan**
Best viewed by-commit. 

Will be merged with rebase.

Part of the #733.

### CODE REVIEW

File: pkg/neonvm/controllers/vm_controller.go [hunk 1]
```diff
@@ -311,6 +311,41 @@ func (r *VMReconciler) updateVMStatusMemory(
 	}
 }
 
+func (r *VMReconciler) acquireOverlayIP(ctx context.Context, vm *vmv1.VirtualMachine) error {
+	if vm.Spec.ExtraNetwork == nil || !vm.Spec.ExtraNetwork.Enable || len(vm.Status.ExtraNetIP) != 0 {
```
Comment (sharnoff) [inline pkg/neonvm/controllers/vm_controller.go:316]: IIUC, this is equivalent to: "if the extra network is disabled OR we already have an extra IP address, nothing to do" , right?

(worth adding that as a comment?)

**Review by sharnoff — APPROVED** [review]
Looks good, but I think d5defa1d7ff24f7f07c94bd31f6e7eefc97f72f9 ("extract createPod") may leak ssh secrets in edge cases. Can give a more thorough example of what I mean later, but in the meantime maybe let's merge the other commits?

**Comment by sharnoff** [comment]
@Omrigan before merging, can you rename the commits to append the PR number?

**Comment by Omrigan** [comment]
> @Omrigan if you're going to rebase-and-merge, could you rename the commits to append the PR number?

Right, totally forgot! I guess it is 3 strikes to automate it, first take is at https://github.com/neondatabase/autoscaling/pull/1152/files.

**Pull request merged**